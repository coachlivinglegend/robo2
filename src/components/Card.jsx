import React from 'react';
import { useLocation, useNavigate } from 'react-router-dom';
const Card = ({ name, email, id }) => {
  const navigate = useNavigate();
  // const a = useNavigate();
  const location = useLocation();
  // const h = useNavigationType();
  // const k = useParams();
  // a: Æ’ (to, options)
  // f: {pathname: '/', search: '', hash: '', state: null, key: 'default'}
  // h: "POP"
  // k: {*: ''}

  return (
    <div
      onClick={() => (location.pathname === '/' ? navigate(`/${name}`) : null)}
      className='bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'
    >
      <img alt='robots' src={`https://robohash.org/${id}test?200x200`} />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  );
};

export default Card;
