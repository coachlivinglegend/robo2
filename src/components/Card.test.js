import { shallow, render, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Card from './Card';

test('expect to render Card Component', () => {
  expect.assertions(1);
  expect(
    shallow(
      <BrowserRouter>
        <Card />
      </BrowserRouter>
    ).length
  ).toEqual(1);
});

test('expect to take a snapshot', () => {
  // eslint-disable-next-line testing-library/render-result-naming-convention
  const wrapper = render(
    <BrowserRouter>
      <Card />
    </BrowserRouter>
  );
  expect(wrapper).toMatchSnapshot();
});
