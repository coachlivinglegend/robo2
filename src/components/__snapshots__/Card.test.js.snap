// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to take a snapshot 1`] = `
<div
  class="bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
>
  <img
    alt="robots"
    src="https://robohash.org/undefinedtest?size=300x300"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
